src/shared/sam/activation.ts

```diff
@@ -27,13 +30,55 @@ import { SamDebugConfigProvider } from './debugger/awsSamDebugger'
 import { addSamDebugConfiguration } from './debugger/commands/addSamDebugConfiguration'
 import { SamDebugSession } from './debugger/samDebugSession'
 import { AWS_SAM_DEBUG_TYPE } from './debugger/awsSamDebugConfiguration'
+import { ext } from '../extensionGlobals'
+
+const YAML_PLUGIN = 'redhat.vscode-yaml'
+const YAML_PROMPT = 'dontShowYamlPromptEver'
 
 /**
  * Activate SAM-related functionality.
  */
 export async function activate(ctx: ExtContext): Promise<void> {
     initializeSamCliContext({ settingsConfiguration: ctx.settings })
 
+    const dontPromptEverAgain = ext.context.globalState.get<boolean>(YAML_PROMPT)
+    // only pop this up in VS Code and Insiders since other VS Code-like IDEs (e.g. Theia) may not have a marketplace or contain the YAML plugin
```

Comment:
Haven't tested but it's defensive. There's also no guarantee that the extension we have tagged exists. We'll reevaluate this in the future as we dig more into that.

Reply:
Not sure I understand. The user could install the extension even if it wasn't in the alternate marketplace by installing the vsix.

If we have a hard dependency on it for this kind of functionality, shouldn't we always show the prompt? Otherwise the user wouldn't know to install it.

Reply:
It's not really a hard dependency in that the extension will still work without it, just in a more manual form. If that's the case, we'll want to have a different prompt since we can't guarantee that the targeted extension will be in whatever marketplace that IDE is using.